jena-extras/jena-querybuilder/src/main/java/org/apache/jena/arq/querybuilder/handlers/DatasetHandler.java

```diff
@@ -54,24 +57,42 @@ public void addAll(DatasetHandler datasetHandler) {
     }
 
     /**
-     * Add a graph name to the from named list.
-     *
-     * @param graphName The graph name to add.
+     * Converts an object into a graph name.
+     * @param graphName the object that represents the graph name.
+     * @return the string that is the graph name.
      */
-    public void fromNamed(String graphName) {
-        query.addNamedGraphURI(graphName);
+    String asGraphName(Object graphName) {
+        // package private for testing access
+        if (graphName instanceof String) {
+            return (String) graphName;
+        }
+        if (graphName instanceof FrontsNode) {
+            return asGraphName(((FrontsNode)graphName).asNode());
+        }
+        if (graphName instanceof Node_URI) {
+            return ((Node_URI)graphName).getURI();
+        }
+        if (graphName instanceof Node_Literal) {
+            return asGraphName(((Node_Literal)graphName).getLiteralValue());
+        }
+
+        return graphName.toString();
     }
 
     /**
-     * Add the graph names to the from named list.
-     *
-     * The names are ordered in as defined in the collection.
+     * Add a graph name to the from named list.
      *
-     * @param graphNames The from names to add.
+     * @param graphName The graph name to add.
      */
-    public void fromNamed(Collection<String> graphNames) {
-        for (String uri : graphNames) {
-            query.addNamedGraphURI(uri);
+    public void fromNamed(Object graphName) {
+        if (graphName instanceof Collection) {
```

Comment:
Coudl be a java17 

```
if (graphName instanceof Collection collection) {
   collection.forEach(this:fromNamed);
}
```

Reply:
Javadoc - yes :-( It can be a bit bumpy.

The build uses Java17 so language features are available if you want. Doesn't requite it all converted. Use, or not, as desired.

The javadoc seems to be built properly.